### PR TITLE
feat(defaults): zS triggers Inspect command

### PR DESCRIPTION
**Pull request by craigmac**
Problem: no map to quickly see what highlights are applied to item under cursor
(A) Solution: map zS in normal mode, vim-scriptease style.

### CODE REVIEW

**Review by gpanders — APPROVED** [review]

File: runtime/doc/syntax.txt [hunk 1]
```diff
@@ -4815,6 +4815,9 @@ You can see all the groups currently active with this command: >
     :so $VIMRUNTIME/syntax/hitest.vim
 This will open a new window containing all highlight group names, displayed
 in their own color.
+                                                        *zS-default*
+You can see a short summary of the the highlight groups applied to a given
+position using the keymapping `zS` in normal mode. |default-mappings|
```
Comment (dundargoc) [inline runtime/doc/syntax.txt:4820]: If Inspect is documented you should reference it IMO. Also this risks duplication with the documentation for inspect.

**Comment by craigmac** [comment]
> do we _really_ need a default keymap for this? 

There's no absolute need no, but it's not overriding a default binding, so I thought I'd put up this idea and gauge interest. I was at least hoping to gain a clearer picture of what can/can't/shouldn't be considered for default-mappings - maybe there's already a consensus somewhere that I missed when looking for it, in that my apologies for wasting your time.

**Comment by gpanders** [comment]
I'm in favor.

* `zS` is unused
* It makes a useful feature (a little) more accesible
* There's precedent from vim-scriptease (I've had the [same mapping configured for a long time](https://github.com/gpanders/dotfiles/blob/6663847837a3c4f8ccd3cf79f712966eaf9204b3/.config/nvim/init.vim#L97-L98))

Not sure what else we'd want to use `zS` for. I'd say we could even take this further and map `z<C-S>` to `:InspectTree` as well (`:InspectTree` is more annoying to enter manually since tab completion first completes `:Inspect`, so it requires a lot more key presses).

**Comment by lewis6991** [comment]
I'm against:
- `:Inspect` is already easy to execute.
- Listing highlights under the cursor is a debug thing, useful for nvim devs and plugins devs, which is very much who vim-scriptease is aimed at. 99% of users won't need this.
- A keymap being currently unused shouldn't be used to justify adding new mappings, that's already a requirement.

The `z` prefix is mostly used for fold and spelling functions, so unless there's a good reason, that it remains for that.

**Review by justinmk — APPROVED** [review]
LGTM after dundar comment is addressed. 

p.s. There are probably many more scriptease mappings / concepts we want to absorb.

> Listing highlights under the cursor is a debug thing, useful for nvim devs and plugins devs,

It's more fundamental than that. Users often wonder what is providing highlights, just like they wonder where a mapping came from. That's also why this is exposed in the mouse right-click menu. Inspecting the editor will become more and more commonplace, especially with the mouse menu. 

It makes sense to treat lower-level "debugging" as more of an opt-in use-case, but that's different than "troubleshooting", which should be zero-friction.

**Comment by lewis6991** [comment]
> just like they wonder where a mapping came from.

This directly implies that we need to add a mapping to find out where a mapping came from. I thought `:map ` was perfectly sufficient for that, just as `:Inspect` is sufficient for finding where highlights come from. Why does the general user need highly efficient access to either of these?

File: runtime/lua/vim/_defaults.lua [hunk 2]
```diff
@@ -65,6 +65,11 @@ do
     desc = ':help CTRL-L-default',
   })
 
+  --- https://github.com/tpope/vim-scriptease style mapping for `:Inspect`.
+  ---
+  --- See |zS-default|
+  vim.keymap.set('n', 'zS', '<Cmd>Inspect<CR>', { desc = ':help zS-default' })
```
Comment (dundargoc) [inline runtime/lua/vim/_defaults.lua:71]: nit/suggestion: shouldn't make a difference, but perhaps more idiomatic?

```suggestion
  vim.keymap.set('n', 'zS', vim.show_pos, { desc = ':help zS-default' })
```

File: runtime/doc/syntax.txt (hunk 1 above)
Comment (craigmac) [reply runtime/doc/syntax.txt]: I've reworded it and included the link to `:Inspect`

**Comment by mfussenegger** [comment]
:100: to what Lewis said and want to add that core using up more and more keymaps by default also makes it harder for plugins to use or recommend mappings.

keymaps are a finite resource which imho should be reserved for actions that are common and frequently used - that's also where they excel due to memorization aspect. Troubleshooting is not something you do all the time.

**Pull request closed without merging**